### PR TITLE
PLATFORM-3645 | Move a part of fandom domain logic to ember-fandom

### PR DESCRIPTION
**Pull request by rogatty**
## Links
* https://wikia-inc.atlassian.net/browse/PLATFORM-3645
* https://wikia-inc.atlassian.net/browse/XW-5167
* https://github.com/Wikia/ember-fandom/pull/9

## Description
Move `instance-initializers/config.js` and `instance-initializers/request-host.js` to ember-fandom. It required some config clean up. I used this occasion to remove redundant code from `services/i18n.js`. I had to remove Crowdin In-context translations code to simplify things but Keiko confirmed it never worked as expected and therefore was not used.

## Who might be interested
@Wikia/x-wing @Wikia/core-platform-team

### CODE REVIEW

**Comment by rybmat** [comment]
@rogatty we had some conflicts with ember-auto-import in ember-fandom and derequire in mobile-wiki and because of that we could not update ember-fandom version. Is this problem resolved?

**Comment by rogatty** [comment]
@rybmat yes, I replaced ember-auto-import with ember-browserify in ember-fandom.

File: app/routes/application.js [hunk 1]
```diff
@@ -1,26 +1,23 @@
-
 import { inject as service } from '@ember/service';
 import Route from '@ember/routing/route';
 import { getOwner } from '@ember/application';
 import { getWithDefault, get } from '@ember/object';
 import Ember from 'ember';
 import { isEmpty } from '@ember/utils';
 import { run } from '@ember/runloop';
+import redirect from '@wikia/ember-fandom/utils/application-redirect';
```
Comment (hakubo) [inline app/routes/application.js]: please import it as applicationRedirect

File: app/utils/errors.js [hunk 2]
```diff
@@ -1,5 +1,8 @@
 import { defineError } from 'ember-exex/error';
 
+// TODO remove redundant errors from this file and use ones from ember-fandom across app
+import errors from '@wikia/ember-fandom/utils/errors';
```
Comment (hakubo) [inline app/utils/errors.js]: import {DontLogMeError} ?

File: app/utils/mediawiki-fetch.js [hunk 3]
```diff
@@ -2,12 +2,12 @@ import fetch from 'fetch';
 import config from '../config/environment';
 
 export default function mediawikiFetch(url, options = {}) {
-	if (typeof FastBoot !== 'undefined' && config.fastbootOnly.mediawikiDomain) {
+	if (typeof FastBoot !== 'undefined' && config.APP.internalCache) {
```
Comment (hakubo) [inline app/utils/mediawiki-fetch.js:5]: yas!

File: package-lock.json [hunk 4]
```diff
@@ -23097,14 +21416,6 @@
       "resolved": "https://registry.npmjs.org/wordwrap/-/wordwrap-0.0.2.tgz",
       "integrity": "sha1-t5Zpu0LstAn4PVg8rVLKF+qhZD8="
     },
-    "worker-farm": {
```
Comment (hakubo) [inline package-lock.json]: preety big package-lock change, are you sure?

Comment (rogatty) [reply package-lock.json]: I think it's because ember-auto-import and its dependencies were removed.

Comment (rogatty) [reply package-lock.json]: Anyway, I have npm 6.1.0 and that's what I got after `npm i` :man_shrugging:

**Review by hakubo — APPROVED** [review]

**Pull request merged**